FlowerShop/Controllers/AccountController.cs

```diff
@@ -0,0 +1,77 @@
+using FlowerShop.DTOs;
+using FlowerShop.Interfaces;
+using FlowerShop.Models;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace FlowerShop.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class AccountController : ControllerBase
+    {
+        private readonly ITokenService _tokenService;
+        private readonly UserManager<AppUser> _userManager;
+
+        public AccountController(UserManager<AppUser> userManager, ITokenService tokenService)
+        {
+            _userManager = userManager;
+            _tokenService = tokenService;
+        }
+
+        [HttpPost("register")] //POST: api/account/register
+        public async Task<ActionResult<UserDto>> Register(RegisterDTO registerDto)
+        {
+            if(await UserExists(registerDto.Username)) return BadRequest();
+
+            var user = new AppUser
+            {
+                UserName = registerDto.Username.ToLower(),
+                Name = registerDto.Name,
+                Surname = registerDto.Surname,
+            };
+            var result = await _userManager.CreateAsync(user, registerDto.Password);
+            
+            if(!result.Succeeded) return BadRequest(result.Errors);
+
+            var roleResult = await _userManager.AddToRoleAsync(user, "Member");
+            
+            if(!roleResult.Succeeded) return BadRequest(result.Errors);
+
+            return new UserDto
+            {
+                Username = user.UserName,
+                Name = user.Name,
+                Surname = user.Surname,
+                Token = await _tokenService.CreateToken(user),
+            };
+        }
+        [HttpPost("login")]
+        public async Task<ActionResult<UserDto>> Login(LoginDTO loginDto)
+        {
+            var user = await _userManager.Users
+            .SingleOrDefaultAsync(x =>x.UserName == loginDto.Username);
+            
+            if(user == null) return Unauthorized("Invalid username");
+            
+            var result = await _userManager.CheckPasswordAsync(user, loginDto.Password);
+
+            if(!result) return Unauthorized("Invalid password");
```

Comment:
For security reasons, I think we shouldn't return which part of the login was wrong and just say that Username or Password is incorrect.

Reply:
Should the previous part also return Wrong username or password too?

Reply:
Yeah both